File: src/test/resources/test/slack-notifier-test.js
var t = require('/lib/xp/testing');

t.mock('/lib/http-client', {
    request: function (params) {
        return {
            "status": 200
        }
    }
});

exports.testNotify = function () {
    var slackNotifier = require('/lib/slack-notifier');

    var config = {
        url: 'url',
        project: 'test',
        reportOnSuccess: false,
        reportOnFailure: false
    };

    slackNotifier.newInstance(config).test('test');
};

exports.testSuccess = function () {
    var slackNotifier = require('/lib/slack-notifier');

    var config = {
        url: 'url',
        project: 'test',
        reportOnSuccess: true,
        reportOnFailure: false
    };

    slackNotifier.newInstance(config).success('SUCCESS');
};

exports.testFailed = function () {
    var slackNotifier = require('/lib/slack-notifier');

    var config = {
        url: 'url',
        project: 'test',
        reportOnSuccess: false,
        reportOnFailure: false
    };

    slackNotifier.newInstance(config).failed('FAILED');
};

File: src/test/java/com/enonic/app/snapshotter/SlackNotifierTest.java
package com.enonic.app.snapshotter;

import org.junit.jupiter.api.Test;

import com.enonic.xp.testing.ScriptTestSupport;

public class SlackNotifierTest
    extends ScriptTestSupport
{

    @Test
    public void testNotify()
    {
        runFunction( "test/slack-notifier-test.js", "testNotify" );
    }

    @Test
    public void testSuccess()
    {
        runFunction( "test/slack-notifier-test.js", "testSuccess" );
    }

    @Test
    public void testFailed()
    {
        runFunction( "test/slack-notifier-test.js", "testFailed" );
    }

}
